### PR TITLE
update parent pom, use plugin bom, remove powermock

### PR DESCRIPTION
**Pull request by jetersen**
Updated to latest plugin parent pom. 
Added plugin bom to help manage plugin versions for other plugins. You can read more about plugin bom here: https://github.com/jenkinsci/bom
Updated jenkins.version according to https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/
Added plugin bom to make it easier to manage the plugin and the dependencies.
Updated JCasC test to use the JCasC test-harness.

replaced all powermock tests with mockito's mockStatic

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jetersen** [comment]
linking https://github.com/jenkinsci/plugin-pom/pull/441 so that this PR is discoverable for anyone else who is struggling with powermock to mockito mockStatic migration 😅

**Comment by offa** [comment]
Related:  https://github.com/jenkinsci/plugin-pom/pull/442

This PR is of particular importance for Java 11+ compatibility.

**Review by offa — APPROVED** [review]

**Comment by jetersen** [comment]
@scaytrase resolved merge conflict :)

**Comment by jetersen** [comment]
This PR should take care of most dependabot PRs 😄

File: pom.xml [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+        <version>4.31</version>
         <relativePath />
     </parent>
     <packaging>hpi</packaging>
     <properties>
         <revision>1.25</revision>
         <changelist>-SNAPSHOT</changelist>
         <!-- Baseline Jenkins version you use to build the plugin. Users must have this version or newer to run. -->
-        <jenkins.version>2.60.3</jenkins.version>
+        <jenkins.version>2.303.1</jenkins.version>
```
Comment (scaytrase) [inline pom.xml:14]: Just a note for myself. This is an LTE relase 5 months old

Comment (jetersen) [reply pom.xml:14]: We can drop it down a little with a matching plugin bom

**Review by scaytrase — APPROVED** [review]
LGTM 👍

**Comment by jetersen** [comment]
https://github.com/jenkinsci/configuration-as-code-plugin/issues/1815 was the reason for this PR to fix plugin loading conflict

**Comment by scaytrase** [comment]
thanks @jetersen !

**Comment by jetersen** [comment]
@scaytrase your welcome!

**Pull request merged**